### PR TITLE
jewel: mon: OSDMonitor: make 'osd crush move ...' work on osds

### PR DESCRIPTION
**Pull request by shinobu-x**
http://tracker.ceph.com/issues/18683

### CODE REVIEW

**Comment by smithfarm** [comment]
@shinobu-x Looks good, thanks.

**Comment by shinobu-x** [comment]
@smithfarm thanks for your comment. Just question, how did you add label?

**Comment by smithfarm** [comment]
@shinobu-x Maybe you don't have sufficient github permissions.

**Comment by shinobu-x** [comment]
@smithfarm o.k, got it. thanks for your answer for my stupid question :)

**Comment by smithfarm** [comment]
Marking DNM because it causes failures in the rados suite. See comment in the code for the fix.

File: qa/workunits/mon/crush_ops.sh [hunk 1]
```diff
@@ -74,11 +74,16 @@ ceph osd crush rm foo
 ceph osd crush rm osd.$o2 host2
 ceph osd crush rm host2
 
+ceph osd crush add-bucket foo host
+ceph osd crush move foo root=default rack=localrack
+
+ceph osd crush create-or-move osd.$o1 1.0 root=default
+ceph osd crush move osd.$o1 host=foo
+ceph osd find osd.$o1 | grep host | grep foo
```
Comment (smithfarm) [inline qa/workunits/mon/crush_ops.sh]: Looks like this is causing rados suite tests to fail like so:

```
2017-02-13T21:59:24.604 INFO:tasks.workunit.client.0.smithi199.stderr:+ ceph osd find osd.7
2017-02-13T21:59:24.658 INFO:teuthology.orchestra.run.smithi201.stderr:2017-02-13 21:59:24.657796 7f0b5cc64700 -1 WARNING: the following dangerous and experimental features are enabled: *
2017-02-13T21:59:24.665 INFO:teuthology.orchestra.run.smithi201.stderr:2017-02-13 21:59:24.663054 7f0b5cc64700 -1 WARNING: the following dangerous and experimental features are enabled: *
2017-02-13T21:59:24.676 INFO:tasks.workunit.client.0.smithi199.stderr:2017-02-13 21:59:24.670512 7efe4a460700 -1 WARNING: the following dangerous and experimental features are enabled: *
2017-02-13T21:59:24.688 INFO:tasks.workunit.client.0.smithi199.stderr:2017-02-13 21:59:24.683239 7efe4a460700 -1 WARNING: the following dangerous and experimental features are enabled: *
2017-02-13T21:59:25.399 INFO:tasks.workunit.client.0.smithi199.stderr:Invalid command:  osd.7 doesn't represent an int
2017-02-13T21:59:25.399 INFO:tasks.workunit.client.0.smithi199.stderr:osd find <int[0-]> :  find osd <id> in the CRUSH map and show its location
2017-02-13T21:59:25.399 INFO:tasks.workunit.client.0.smithi199.stderr:Error EINVAL: invalid command
2017-02-13T21:59:25.410 INFO:tasks.workunit:Stopping ['mon/pool_ops.sh', 'mon/crush_ops.sh', 'mon/osd.sh', 'mon/caps.sh'] on client.0...
```

@shinobu-x Please try to change this line to

```
ceph osd find $o1 | grep host | grep foo
```

and add a line like this at the end of the commit message:

```
Conflicts:
    qa/workunits/mon/crush_ops.sh: adapted "ceph osd find" to jewel syntax
```

Comment (shinobu-x) [inline qa/workunits/mon/crush_ops.sh]: @smithfarm Very sorry for the late =) I will give it a shot.

**Comment by smithfarm** [comment]
@shinobu-x Almost there! Now just squash the two commits together and force-push the branch so the changes appear here in the PR.

**Comment by shinobu-x** [comment]
@smithfarm Got it. will do!

**Comment by shinobu-x** [comment]
jenkins test this please

**Comment by shinobu-x** [comment]
@smithfarm A new commit was squashed into previous one. Please check.

**Comment by shinobu-x** [comment]
Nothing was squashed properly =)

**Comment by shinobu-x** [comment]
@smithfarm Please check :)

**Comment by smithfarm** [comment]
**Note: this PR is targeting the 10.2.8 release - not to be merged until v10.2.7 tag is added!**

@liewegas @jdurgin This PR was included in the rados run at http://tracker.ceph.com/issues/19538#note-4 which had the following results:

Of 227 total jobs:

* 222 passed on first try
* 3 passed on second try
* remaining 2 are problematic

(see http://tracker.ceph.com/issues/19538#note-4 for details)

Do you think this run is sufficient to merge this PR once the v10.2.7 tag has been added?

**Pull request merged**